cfme/configure/configuration/analysis_profile.py

```diff
@@ -0,0 +1,440 @@
+
+    def delete(self, cancel=False):
+        """Delete self via details page"""
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select("Delete this Analysis Profile",
+                                               handle_alert=not cancel)
+        view = self.create_view(
+            AnalysisProfileDetailsView if cancel else AnalysisProfileAllView)
+        view.flush_widget_cache()
+        assert view.is_displayed
+        if not cancel:
+            view.flash.assert_success_message('Analysis Profile "{}": Delete successful'
+                                              .format(self.description))
+        else:
+            assert view.flash.messages == []
+
+    def copy(self, new_name=None, cancel=False):
+        """Copy the Analysis Profile"""
+        # Create a new object to return in addition to running copy in the UI
+        if not new_name:
+            new_name = self.name + "-copy"
+        profile_args = self.__dict__.copy()
+        profile_args['name'] = new_name
+        new_profile = AnalysisProfile(**profile_args)
+
+        # actually run copy in the UI, fill the form
+        view = navigate_to(self, 'Copy')
+        form_args = self.form_fill_args(updates=profile_args)
+        view.form.fill(form_args)
+        if cancel:
+            view.cancel.click()
+        else:
+            view.add.click()
+
+        # check the result
+        view = self.create_view(
+            AnalysisProfileDetailsView if cancel else AnalysisProfileAllView)
+        view.flush_widget_cache()
+        assert view.is_displayed
+        view.flash.assert_no_error()
+        view.flash.assert_success_message(
+            'Add of new Analysis Profile was cancelled by the user'  # yep, not copy specific
+            if cancel
+            else 'Analysis Profile "{}" was saved'.format(new_profile.name))
+
+        return new_profile
+
+    @property
+    def exists(self):
+        try:
+            navigate_to(self, 'Details')
+        except (NavigationDestinationNotFound, CandidateNotFound):
+            return False
+        else:
+            return True
+
+    def as_fill_value(self):
+        """String representation of an Analysis Profile in CFME UI"""
+        return self.name
+
+    def form_fill_args(self, updates=None):
+        """Build a dictionary of nested tab_forms for assoc_fill from a flat object dictionary
+        If updates dictionary is passed, it is used instead of `self`
+        """
+        # well this is ugly. but works for create and update
+        # when updates is dict, getattr results in errors, using the default breaks update
+        # when updates is None, calling get results in an error because its None.get
+        # TODO FIXME
+        # For some reason the dict copy doesn't seem to be working here? its updating self.attr
+        # when popping the assoc_column item instead of the new dict.
```

Comment:
I'll toy with copying the entire object instead of just copying the its `__dict__`.